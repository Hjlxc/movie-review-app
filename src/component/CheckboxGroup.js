import React from 'react';
import { Checkbox } from 'antd';

import { VerticalFlexWrapper } from '../container/styledComponent';

export default function({ options, checked, onOptionClick }) {
  return (
    <VerticalFlexWrapper>
      {options.map((option) => (
        <div key={option}>
          <Checkbox
            onChange={() => onOptionClick(option)}
            checked={checked[option]}
          >
            {option}
          </Checkbox>
        </div>
      ))}
    </VerticalFlexWrapper>
  );
}
