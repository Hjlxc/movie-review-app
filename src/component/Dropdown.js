import React from 'react';

import { Menu, Dropdown } from 'antd';
import { DownOutlined } from '@ant-design/icons';

export default function({ select, options, onChange }) {
  const menu = (
    <Menu onClick={onChange}>
      {options.map((option) => (
        <Menu.Item key={option}>{option}</Menu.Item>
      ))}
    </Menu>
  );

  return (
    <Dropdown overlay={menu}>
      <a
        className="ant-dropdown-link"
        href="/"
        onClick={(e) => e.preventDefault()}
      >
        {select} <DownOutlined />
      </a>
    </Dropdown>
  );
}
