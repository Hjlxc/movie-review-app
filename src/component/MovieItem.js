import React from 'react';

import { Rate } from 'antd';

import {
  MovieItemWrapper,
  MovieItemSwatch,
  BoldWrapper,
} from '../container/styledComponent';

export default function MovieItem({ swatch, title, rating, onClick }) {
  return (
    <MovieItemWrapper>
      <MovieItemSwatch src={swatch} onClick={onClick} alt={`${title} Poster`} />
      <BoldWrapper>{title}</BoldWrapper>
      <Rate allowHalf disabled value={rating} />
    </MovieItemWrapper>
  );
}
