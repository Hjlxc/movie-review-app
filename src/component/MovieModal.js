import React from 'react';
import { Modal, Progress, Tooltip } from 'antd';

import {
  HorizontalFlexWrapper,
  MovieModalSwatch,
  VerticalFlexWrapper,
  BoldWrapper,
  TextWrapper,
  MovieModalSwatchWrapper,
  Gap,
} from '../container/styledComponent';

export default function MovieModal({
  swatch,
  title,
  vote_average,
  vote_count,
  overview,
  release_date,
  original_language,
  visible,
  onClose,
}) {
  return (
    <Modal
      title={title}
      centered
      visible={visible}
      footer={null}
      onCancel={onClose}
    >
      <HorizontalFlexWrapper>
        <MovieModalSwatchWrapper>
          <MovieModalSwatch src={swatch} alt={`${title} Poster`} />
        </MovieModalSwatchWrapper>
        <VerticalFlexWrapper>
          <HorizontalFlexWrapper>
            <TextWrapper>{release_date}</TextWrapper>
            <Gap />
            <TextWrapper>{`Language: ${original_language}`}</TextWrapper>
          </HorizontalFlexWrapper>
          <div>
            <Tooltip placement="right" title={`${vote_count} Voting`}>
              <Progress
                width={40}
                type="circle"
                percent={vote_average * 10}
                strokeColor={getColorGradient(vote_average)}
              />
            </Tooltip>
          </div>
          <BoldWrapper>Overview</BoldWrapper>
          <TextWrapper>{overview}</TextWrapper>
        </VerticalFlexWrapper>
      </HorizontalFlexWrapper>
    </Modal>
  );
}

// get color gradient based on rating
// 0-10 -> red-green
export const getColorGradient = (rating) => {
  const ratio = rating / 10;
  return `rgb(${255 - 255 * ratio},${255 * ratio},0)`;
};
