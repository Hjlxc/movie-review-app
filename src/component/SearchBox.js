import React from 'react';
import { Input } from 'antd';

export default function({ placeholder, onSearch }) {
  return (
    <Input.Search placeholder={placeholder} onSearch={onSearch} enterButton />
  );
}
