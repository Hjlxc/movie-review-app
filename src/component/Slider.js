import React from 'react';
import { Slider } from 'antd';

import { VerticalFlexWrapper, BoldWrapper } from '../container/styledComponent';

export default function({ title, min, max, selectMin, selectMax, onChange }) {
  const marks = { [min]: min, [max]: max };
  return (
    <VerticalFlexWrapper>
      {title && <BoldWrapper>{title}</BoldWrapper>}
      <Slider
        min={min}
        max={max}
        range
        defaultValue={[selectMin, selectMax]}
        step={0.1}
        onChange={onChange}
        marks={marks}
      />
    </VerticalFlexWrapper>
  );
}
