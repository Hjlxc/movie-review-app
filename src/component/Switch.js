import React from 'react';
import { Switch } from 'antd';

export default function({ checked, onChange }) {
  return <Switch checked={checked} onChange={onChange} />;
}
