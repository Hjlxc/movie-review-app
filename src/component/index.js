import CheckboxGroup from './CheckboxGroup';
import Slider from './Slider';
import MovieItem from './MovieItem';
import MovieModal from './MovieModal';
import Switch from './Switch';
import SearchBox from './SearchBox';

export { CheckboxGroup, Slider, MovieItem, MovieModal, Switch, SearchBox };
