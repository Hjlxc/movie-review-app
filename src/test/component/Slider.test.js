import React from 'react';
import renderer from 'react-test-renderer';

import { Slider } from '../../component';

describe('Test Switch Component', () => {
  let testRenderer;
  let instance;
  const mockOnChange = jest.fn();
  beforeAll(() => {
    testRenderer = renderer.create(
      <Slider
        min={1}
        max={5}
        selectMin={2}
        selectMax={4}
        onChange={mockOnChange}
      />
    );
    instance = testRenderer.root;
  });

  it('Test switch render with correct props', () => {
    expect(instance.props.min).toBe(1);
    expect(instance.props.max).toBe(5);
    expect(instance.props.selectMin).toBe(2);
    expect(instance.props.selectMax).toBe(4);
  });

  it('Test onChange Function', () => {
    instance.props.onChange();
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
