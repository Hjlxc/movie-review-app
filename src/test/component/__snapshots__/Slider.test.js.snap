// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Switch Component matches snapshot 1`] = `
<div
  className="ant-slider ant-slider-with-marks"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="ant-slider-rail"
    style={Object {}}
  />
  <div
    className="ant-slider-track ant-slider-track-1"
    style={
      Object {
        "left": "25%",
        "right": "auto",
        "width": "50%",
      }
    }
  />
  <div
    className="ant-slider-step"
  >
    <span
      className="ant-slider-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    />
    <span
      className="ant-slider-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    />
  </div>
  <div
    aria-disabled={false}
    aria-valuemax={5}
    aria-valuemin={1}
    aria-valuenow={2}
    className="ant-slider-handle ant-slider-handle-1"
    onBlur={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="slider"
    style={
      Object {
        "left": "25%",
        "right": "auto",
        "transform": "translateX(-50%)",
      }
    }
    tabIndex={0}
  />
  <div
    aria-disabled={false}
    aria-valuemax={5}
    aria-valuemin={1}
    aria-valuenow={4}
    className="ant-slider-handle ant-slider-handle-2"
    onBlur={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="slider"
    style={
      Object {
        "left": "75%",
        "right": "auto",
        "transform": "translateX(-50%)",
      }
    }
    tabIndex={0}
  />
  <div
    className="ant-slider-mark"
  >
    <span
      className="ant-slider-mark-text"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "left": "0%",
          "msTransform": "translateX(-50%)",
          "transform": "translateX(-50%)",
        }
      }
    >
      1
    </span>
    <span
      className="ant-slider-mark-text"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "left": "100%",
          "msTransform": "translateX(-50%)",
          "transform": "translateX(-50%)",
        }
      }
    >
      5
    </span>
  </div>
</div>
`;
